dotnet/SimpleSauce.Test/SauceOptionsTests.cs

```diff
@@ -132,5 +133,190 @@ public void AcceptsAllSauceLabsValues()
+        {
+            var options = new ChromeOptions();
+            SauceOptions = new SauceOptions(options);
+
+            SauceOptions.BrowserName.Should().BeEquivalentTo(Browser.Chrome);
+            SauceOptions.SeleniumOptions.Should().BeEquivalentTo(options);
+        }
+        [TestMethod]
+        public void AcceptsEdgeOptionsClass()
+        {
+            var options = new EdgeOptions();
+            SauceOptions = new SauceOptions(options);
+
+            SauceOptions.BrowserName.Should().Be(Browser.Edge);
+            SauceOptions.SeleniumOptions.Should().BeEquivalentTo(options);
+        }
+
+        [TestMethod]
+        public void AcceptsFirefoxOptionsClass()
+        {
+            var options = new FirefoxOptions();
+            SauceOptions = new SauceOptions(options);
+
+            SauceOptions.BrowserName.Should().BeEquivalentTo(Browser.Firefox);
+            SauceOptions.SeleniumOptions.Should().BeEquivalentTo(options);
+        }
+
+        [TestMethod]
+        public void AcceptsInternetExplorerOptionsClass()
+        {
+            var options = new InternetExplorerOptions();
+            SauceOptions = new SauceOptions(options);
+
+            SauceOptions.BrowserName.Should().BeEquivalentTo(Browser.IE);
+            SauceOptions.SeleniumOptions.Should().BeEquivalentTo(options);
+        }
+
+        [TestMethod]
+        public void AcceptsSafariOptionsClass()
+        {
+            var options = new SafariOptions();
+            SauceOptions = new SauceOptions(options);
+
+            SauceOptions.BrowserName.Should().BeEquivalentTo(Browser.Safari);
+            SauceOptions.SeleniumOptions.Should().BeEquivalentTo(options);
+        }
+
+        [TestMethod]
+        public void CreatesDefaultBuildName()
+        {
+            Environment.SetEnvironmentVariable("BUILD_TAG", "Not Empty");
+            Environment.SetEnvironmentVariable("BUILD_NAME", "TEMP BUILD");
+            Environment.SetEnvironmentVariable("BUILD_NUMBER", "11");
+
+            SauceOptions.BuildName.Should().BeEquivalentTo("TEMP BUILD: 11");
+        }
+
+        [TestMethod]
+        public void ParsesCapabilitiesFromW3CValues()
+        {
+            SauceOptions.BrowserName = Browser.Firefox;
+            SauceOptions.PlatformName = Platforms.MacOsHighSierra;
+            SauceOptions.BrowserVersion = "77";
+            SauceOptions.PageLoadStrategy = PageLoadStrategy.Eager;
+            SauceOptions.AcceptInsecureCerts = true;
+            SauceOptions.SetWindowRect = true;
+            SauceOptions.Timeout.Implicit = 4;
+            SauceOptions.Timeout.PageLoad = 44;
+            SauceOptions.Timeout.Script = 33;
+            SauceOptions.BuildName = "Build Name";
+
+            var timeouts = new Dictionary<string, int>
+            {
+                { "Implicit", 4 },
+                { "PageLoad", 44 },
+                { "Script", 33 }
+            };
+
+            var expectedCapabilities = new Dictionary<string, object>
```

Comment:
@joshmgrant @titusfortner I'm struggling here, are these the right `expectedCapabilities` for my test? Because I can see that when I do ```DriverOptions SeleniumOptions = new FirefoxOptions()``` I get back. 

```
SeleniumOptions
{{
  "browserName": "firefox",
  "moz:firefoxOptions": {}
}}
    AcceptInsecureCertificates: null
    BrowserExecutableLocation: null
    BrowserName: "firefox"
    BrowserVersion: null
    LogLevel: Default
    PageLoadStrategy: Default
    PlatformName: null
    Profile: null
    Proxy: null
    UnhandledPromptBehavior: Default
    UseLegacyImplementation: false
    UseStrictFileInteractability: false
```
So then the problem is that I have 2 properties on the SeleniumOptions objects. One called `browserName` that comes from setting FirefoxOptions and the other one is `BrowserName` which seems like another property on the `DriverOptions` class from Selenium.

Anyone know what's the deal here?